Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +9,128 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            string personWithHighestBalanceEver = "not found";
+            double? highestBalance = null;
+
+            foreach (string personInformation in peopleAndBalances)
+            {
+                string[] personInformationSplit = personInformation.Split(", ");
+
+                for (int i = 1; i <= personInformationSplit.Length - 1; i++)
+                {
+                    bool isNumber = double.TryParse(personInformationSplit[i], out double balance);
+
+                    if (!isNumber)
+                    {
+                        break;
+                    }
+                    else if (highestBalance == null || balance > highestBalance)
+                    {
+                        highestBalance = balance;
+                        personWithHighestBalanceEver = $"{personInformationSplit[0]} (current balance: {personInformationSplit[^1]})";
+                    }
+                }
+
+            }
+            string informationOnHighestBalance = $"{personWithHighestBalanceEver} had the biggest historic balance";
+
+            return informationOnHighestBalance;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            string personWithBiggestLoss = "not found";
+            double? biggestLoss = null;
+
+            foreach (string personInformation in peopleAndBalances)
+            {
+                string[] personInformationSplit = personInformation.Split(", ");
+
+                for (int i = 1; i <= personInformationSplit.Length - 2; i++)
```

Comment:
You could use a constant to indicate the reason why the loop starts at 1. For example: 
```
const int firstBalanceIndex = 1;
```